ikarus/finiteelements/mechanics/kirchhoffloveshell.hh

```diff
@@ -34,22 +34,23 @@ namespace Ikarus {
  * \tparam B The basis type for the finite element.
  * \tparam FER The type representing the requirements for finite element calculations.
  * \tparam useEigenRef A boolean indicating whether to use Eigen references for efficiency.
+ * \tparam useFlat A boolean indicating if the type of the underlying basis is of the flat or the untouched version.
  */
-template <typename B, typename FER = FERequirements<>, bool useEigenRef = false>
-class KirchhoffLoveShell : public PowerBasisFE<B>,
-                           public Volume<KirchhoffLoveShell<B, FER, useEigenRef>, FETraits<B, FER, useEigenRef>>,
-                           public Traction<KirchhoffLoveShell<B, FER, useEigenRef>, FETraits<B, FER, useEigenRef>>
+template <typename B, typename FER = FERequirements<>, bool useEigenRef = false, bool useFlat = true>
+class KirchhoffLoveShell
+    : public FEBase<B, useFlat>,
+      public Volume<KirchhoffLoveShell<B, FER, useEigenRef, useFlat>, FETraits<B, useFlat, FER, useEigenRef>>,
+      public Traction<KirchhoffLoveShell<B, FER, useEigenRef, useFlat>, FETraits<B, useFlat, FER, useEigenRef>>
 {
 public:
-  using Traits            = FETraits<B, FER, useEigenRef>;
+  using Traits            = FETraits<B, useFlat, FER, useEigenRef>;
```

Comment:
This simply should be `using Traits = FEBase<B, useFlat>::Traits;`

Reply:
If we do it like this, `useEigenRef` is not used at all?

Reply:
Ah i made a small mistake but yes that the point all the arguments should be forwarded to `FEBase<B, useFlat,FER,useEigenRef>; and all the other elements just use the traits from the base class. Therefore, all the arguments should be passed, I don't know why you left the arguments as default.